app/domain/authentication/authn_oidc/authentication_service.rb

```diff
@@ -0,0 +1,95 @@
+require 'uri'
+require 'openid_connect'
+
+module Authentication
+  module AuthnOidc
+    class AuthenticationService
```

Comment:
I know you were told to look at the K8s authenticator as a model but we just finished doing a huge refactor that might serve as a better model. I would suggest taking a look (it's in master now) and restructuring to be more in line with that.

Reply:
Thanks @jtuttle ! What for example are you referring to?

Reply:
I added a ticket for this, we'll get to it after the split

Reply:
@jtuttle @orenbm @jonahx I understand this can be done after merge to master, right?